test/functional/tests/fault_injection/test_fault_injection_interrupts.py

```diff
@@ -0,0 +1,542 @@
+        core_part = core_dev.partitions[0]
+
+    for _ in TestRun.iteration(range(iterations_per_config),
+                               f"Reload cache configuration {iterations_per_config} times."):
+
+        with TestRun.step("Start cache."):
+            cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+        with TestRun.step("Set cleaning policy to NOP."):
+            cache.set_cleaning_policy(CleaningPolicy.nop)
+
+        with TestRun.step(f"Add core device with {filesystem} filesystem and mount it."):
+            core_part.create_filesystem(filesystem)
+            core = cache.add_core(core_part)
+            core.mount(mount_point)
+
+        with TestRun.step(f"Create test file in mount point of exported object."):
+            test_file = create_test_file()
+
+        with TestRun.step("Check md5 sum of test file."):
+            tf_md5s_bf = test_file.md5sum()
+
+        with TestRun.step(
+                "Get number of dirty data on exported object before core removal interruption."):
+            os_utils.sync()
+            os_utils.drop_caches(DropCachesMode.ALL)
+            cache_drtblcks_before = cache.get_dirty_blocks()
+
+        with TestRun.step("Unmount core."):
+            core.unmount()
+
+        with TestRun.step("Start removing core device."):
+            flush_pid = TestRun.executor.run_in_background(
+                str(cli.remove_core_cmd(str(cache.cache_id), str(core.core_id))))
+            sleep(1)
+
+        with TestRun.step("Interrupt core removing"):
+            percentage = casadm_parser.get_flushing_progress(int(cache.cache_id), int(core.core_id))
+            while percentage < 50:
+                percentage = casadm_parser.get_flushing_progress(
+                    int(cache.cache_id), int(core.core_id))
+            TestRun.executor.run(f"kill -s SIGINT {flush_pid}")
+
+        with TestRun.step(
+                "Check number of dirty data on exported object after core removal interruption."):
+            cache_drtblcks_after = cache.get_dirty_blocks()
+            if cache_drtblcks_after >= cache_drtblcks_before:
+                TestRun.LOGGER.error("Quantity of dirty lines after core removal interruption "
+                                     "should be lower.")
+            if int(cache_drtblcks_after) == 0:
+                TestRun.LOGGER.error("Quantity of dirty lines after core removal interruption "
+                                     "should not be zero.")
+
+        with TestRun.step("Remove core from cache."):
+            core.remove_core()
+
+        with TestRun.step("Mount core device."):
+            core_part.mount(mount_point)
+
+        with TestRun.step("Flush cache."):
+            cache.flush_cache()
+
+        with TestRun.step("Check md5 sum of test file again."):
+            if tf_md5s_bf != test_file.md5sum():
+                TestRun.LOGGER.error(
+                    "Md5 sums before and after interrupting core removal are different.")
```

Comment:
Test would be more valuable if cache was stopped, and md5 was checked on core disk. Currently even if all data is clean, it is still cached, so you don't check if file was properly written to core.